goal_src/jak2/engine/game/task/task-arrow.gc

```diff
@@ -5,5 +5,371 @@
 ;; name in dgo: task-arrow
 ;; dgos: ENGINE, GAME
 
+;; +++task-arrow-flags
+(defenum task-arrow-flags
+  :type uint32
+  :bitfield #t
+  (task-arrow-flag-00 0)
+  (task-arrow-flag-01 1)
+  (task-arrow-flag-02 2)
+  (task-arrow-flag-03 3)
+  )
+;; ---task-arrow-flags
+
 ;; DECOMP BEGINS
 
+(let ((a0-0 (new 'static 'skeleton-group
+              :name "skel-task-arrow"
+              :extra #f
+              :info #f
+              :art-group-name "ctywide-arrow"
+              :bounds (new 'static 'vector :w 16384.0)
+              :texture-level #x6
+              :version #x7
+              )
+            )
+      )
+  (set! (-> a0-0 jgeo) 0)
+  (set! (-> a0-0 janim) 2)
+  (set! (-> a0-0 mgeo 0) 1)
+  (set! (-> a0-0 lod-dist 0) 4095996000.0)
+  (add-to-loading-level a0-0)
+  )
+
+(deftype task-arrow-params (structure)
+  ((flags    task-arrow-flags         :offset-assert   0)
+   (map-icon uint16                   :offset-assert   4)
+   (pos      vector           :inline :offset-assert  16)
+   (quat     quaternion       :inline :offset-assert  32)
+   )
+  :method-count-assert 9
+  :size-assert         #x30
+  :flag-assert         #x900000030
+  )
+
+
+(deftype task-arrow (process-drawable)
+  ((pos              vector             :inline :offset-assert 208)
+   (theta            float                      :offset-assert 224)
+   (phi              float                      :offset-assert 228)
+   (dist             float                      :offset-assert 232)
+   (smoothed-dist    float                      :offset-assert 236)
+   (max-dist         float                      :offset-assert 240)
+   (flags            task-arrow-flags           :offset-assert 244)
+   (map-icon         uint16                     :offset-assert 248)
+   (minimap          connection-minimap         :offset-assert 252)
+   (hud-dist         handle                     :offset-assert 256)
+   (base-quat        quaternion         :inline :offset-assert 272)
+   (rod-of-god-scale float                      :offset-assert 288)
+   (moving           symbol                     :offset-assert 292)
+   )
+  :heap-base #xb0
+  :method-count-assert 25
+  :size-assert         #x128
+  :flag-assert         #x1900b00128
+  (:methods
+    (idle () _type_ :state 20)
+    (die () _type_ :state 21)
+    (leave () _type_ :state 22)
+    (task-arrow-method-23 (_type_ vector) none 23)
+    (draw-arrow (_type_) none :behavior task-arrow 24)
+    )
+  )
+
+
+;; WARN: Return type mismatch int vs none.
+(defmethod deactivate task-arrow ((obj task-arrow))
+  (send-event (handle->process (-> obj hud-dist)) 'hide-and-die)
+  ((method-of-type process-drawable deactivate) obj)
+  0
+  (none)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod task-arrow-method-23 task-arrow ((obj task-arrow) (arg0 vector))
```

Comment:
oh, I wonder if this function is for the giant yellow light thing. It might do this check to see the height of the ground, so it stops drawing once it hits something.

Reply:
oh you know what, that probably makes more sense than the arrow in the demo like i thought -- since the code is related to the minimap / the "rod-of-god" etc.  I guess they just left the name alone perhaps? It's certainly not an arrow anymore atleast.